packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -229,11 +271,15 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
 
   @override
   Widget build(BuildContext context) {
-    final Color effectiveActiveColor = widget.activeColor
-      ?? CupertinoColors.activeBlue;
-    final Color? inactiveColor = widget.inactiveColor;
-    final Color effectiveInactiveColor = inactiveColor
-      ?? CupertinoColors.inactiveGray;
+     // Colors need to be resolved in selected and non selected states separately
+    final Set<WidgetState> activeStates = states..add(WidgetState.selected);
+    final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
+
+    final Color activeColor = widget.activeColor ?? CupertinoColors.activeBlue;
+    final Color effectiveActiveColor = widget.fillColor?.resolve(activeStates) ?? activeColor;
```

Comment:
Shall we combine the two lines? 
```suggestion
    final Color activeColor = widget.fillColor?.resolve(activeStates) 
        ?? widget.activeColor
        ?? CupertinoColors.activeBlue;
```

Also for the inactive color below.